packages/syft/src/syft/core/tensor/autodp/row_entity_phi.py

```diff
@@ -299,12 +306,69 @@ def repeat(
                 "'axis' arg is negative and strangely large... not sure what to do."
             )
 
-    def reshape(self, *shape: List[int]) -> RowEntityPhiTensor:
+    def flatten(self, order: Optional[str] = "C") -> RowEntityPhiTensor:
+        new_list = list()
+        for tensor in self.child:
+            new_list.append(tensor.flatten(order))
+        return RowEntityPhiTensor(rows=new_list, check_shape=False)
+
+    def ravel(self, order: Optional[str] = "C") -> RowEntityPhiTensor:
+        new_list = list()
+        for tensor in self.child:
+            new_list.append(tensor.ravel(order))
+        return RowEntityPhiTensor(rows=new_list, check_shape=False)
+
+    def swapaxes(self, axis1: int, axis2: int) -> RowEntityPhiTensor:
+        if axis1 == 0 or axis2 == 0:
+            raise Exception(
+                "For now, you can't swap the first axis b/c that would "
+                "probably create a Gamma Tensor. Sorry about that!"
+            )
+        new_list = list()
+        for tensor in self.child:
+            # Axis=0 for REPT.child is Axis=1 for REPT, so subtract 1
+            new_list.append(tensor.swapaxes(axis1 - 1, axis2 - 1))
+        return RowEntityPhiTensor(rows=new_list, check_shape=False)
+
+    def squeeze(
+        self, axis: Optional[Union[int, Tuple[int, ...]]] = None
+    ) -> RowEntityPhiTensor:
+        if axis == 0:
+            # If the first axis can be squeezed then there is only one
+            # tensor in the REPT, as such it might be a SEPT
+            # TODO: Check if the output type is still a REPT
+            # if isinstance(self.child[0], SEPT): return self.child[0]
+            return RowEntityPhiTensor(rows=self.child[0])
+        else:
+            new_list = list()
+            for tensor in self.child:
+                new_list.append(tensor.squeeze(axis))
+            self.child = new_list
+
+        return RowEntityPhiTensor(rows=new_list, check_shape=False)
+
+    def reshape(
+        self,
+        shape: Union[
+            int,
+            Union[Sequence[int], Sequence[Sequence[int]]],
+        ],
+    ) -> RowEntityPhiTensor:
+        if isinstance(shape, int):
+            raise Exception(
+                f"{type(self)}.reshape for shape: int {shape} is not implemented"
+            )
+        # This is to fix the bug where shape = ([a, b, c], )
+        if isinstance(shape[0], Sequence):
```

Comment:
This was complaining because the shape signature was promising Union int and Sequence not a possible nested sequence... 🙄 yup this stuff can be annoying.